components/libbuild-protocols.rs

```diff
@@ -48,7 +27,7 @@ mod protocols {
                 continue;
             }
             if file.metadata().unwrap().is_file() {
-                files.push(file.path());
+                files.push(file.path().to_string_lossy().into_owned());
```

Comment:
Seem like something has gone horribly wrong here if we somehow have invalid unicode in `file.path()`. I'd be more comfortable with
```rust
                  files.push(file.path().to_str().unwrap().into());
```